modules/photon-cdn.php

```diff
@@ -113,10 +114,32 @@ public static function fix_script_relative_path( $relative, $src ) {
 
 		// We only treat URLs that have wp-includes in them. Cases like language textdomains
 		// can also use this filter, they don't need to be touched because they are local paths.
-		if ( false === $strpos ) {
-			return $relative;
+		if ( false !== $strpos ) {
+			return substr( $src, 1 + $strpos );
+		}
+
+		// Reverse cdnize_plugin_assets for loading the path
+		if ( preg_match( '#' . preg_quote( self::CDN, '#' ) . 'p/[^/]+/[^/]+/(.*)$#', $src, $m ) ) {
+			return $m[1];
+		}
+
+		return $relative;
+	}
+
+	/**
+	 * Ensure use of the correct relative path when loading the JavaScript translation file.
+	 *
+	 * @param string $file   The path that's going to be loaded.
+	 * @param string $handle The script handle.
+	 * @param string $domain The text domain.
+	 * @return string The transformed local languages path.
+	 */
+	public static function fix_local_script_translation_path( $file, $handle, $domain ) {
+		switch ( $handle ) {
+			case 'jetpack-blocks-editor':
+				return str_replace( 'wp-content/languages/jetpack', 'wp-content/languages/plugins/jetpack', $file );
```

Comment:
The hairy thing here is that we need to rewrite only the plugin translations. It looks like core assumes that these CDN-ed plugins have their translations in the core folder and not in the plugins folder.